core/src/main/java/org/apache/accumulo/core/spi/scan/ScanDispatcher.java

```diff
@@ -115,13 +90,7 @@ public interface DispatchParameters {
    *
    * @since 2.1.0
    */
-
   default ScanDispatch dispatch(DispatchParameters params) {
-    String executor = dispatch((DispatchParmaters) params);
-    if (executor.equals(DefaultScanDispatch.DEFAULT_SCAN_DISPATCH.getExecutorName())) {
-      return DefaultScanDispatch.DEFAULT_SCAN_DISPATCH;
-    }
-
-    return ScanDispatch.builder().setExecutorName(executor).build();
+    return DefaultScanDispatch.DEFAULT_SCAN_DISPATCH;
```

Comment:
I would like feedback on this particular change, because I'm not sure if this should be the right default implementation.

Reply:
I think we need @keith-turner to look at this. But, from what I can tell, this looks ok.

Reply:
All the ITs passed. I will merge as-is, but would still like @keith-turner 's perspective on that particular change when he's able to provide it.

Reply:
The default method only existed for deprecation and compatibility purposes.  If the interface were created from scratch without this history, then the interface method would have no default implementation because that method must be implemented.  If someone wants some sort of default behavior then they should use the default implementation of this interface.  Not sure about the compatibility aspects, but the cleanest form of this method would be the method declaration with no default implementation, that clearly informs people at compile time that they need to implement the method.  @ctubbsii my suggestion would be to remove the default method body from the interface.

Reply:
I created #3263 for this